adapters/rest-adapter-generator/src/main/java/uk/gov/justice/services/adapters/rest/generator/JaxRsImplementationGenerator.java

```diff
@@ -0,0 +1,348 @@
+     * @param component specifies the type of framework {@link Component} that is being implemented
+     * @return a list of {@link TypeSpec} that represent the implementation classes
+     */
+    List<TypeSpec> generateFor(final Collection<Resource> resources, final Component component) {
+        return resources.stream()
+                .map(resource -> createClassFor(resource, component))
+                .collect(Collectors.toList());
+    }
+
+    /**
+     * Create an implementation class for the specified {@link Resource}
+     *
+     * @param resource  the resource to generate as an implementation class
+     * @param component specifies the type of framework {@link Component} that is being implemented
+     * @return a {@link TypeSpec} that represents the implementation class
+     */
+    private TypeSpec createClassFor(final Resource resource, final Component component) {
+        final TypeSpec.Builder classSpecBuilder = classSpecFor(resource, component);
+
+        resource.getActions().values().forEach(action -> {
+            classSpecBuilder.addField(dispatcherFieldFor(action));
+            classSpecBuilder.addMethods(forEach(action));
+        });
+
+        return classSpecBuilder.build();
+    }
+
+    /**
+     * Creates a {@link TypeSpec.Builder} from an initial template of an implementation class
+     *
+     * @param resource  the resource to generate as an implementation class
+     * @param component specifies the type of framework {@link Component} that is being implemented
+     * @return a {@link TypeSpec.Builder} that represents the implementation class
+     */
+    private TypeSpec.Builder classSpecFor(final Resource resource, final Component component) {
+        return classBuilder("Default" + resourceInterfaceNameOf(resource))
+                .addSuperinterface(interfaceClassNameFor(resource))
+                .addModifiers(PUBLIC)
+                .addAnnotation(Stateless.class)
+                .addAnnotation(AnnotationSpec.builder(Adapter.class)
+                        .addMember(DEFAULT_ANNOTATION_PARAMETER, "$T.$L", Component.class, component)
+                        .build())
+                .addField(FieldSpec.builder(RestProcessor.class, "restProcessor")
+                        .addAnnotation(Inject.class)
+                        .build())
+                .addField(FieldSpec.builder(HttpHeaders.class, "headers")
+                        .addAnnotation(Context.class)
+                        .build());
+    }
+
+    /**
+     * Process the body or bodies for each action.
+     *
+     * @param action the action to process
+     * @return the list of {@link MethodSpec} that represents each method for the action
+     */
+    private List<MethodSpec> forEach(final Action action) {
+        final boolean actionHasNoBody = !action.hasBody();
+
+        if (actionHasNoBody) {
+            return processNoActionBody(action);
```

Comment:
move Collections.SingletonList up to here as it is this method that requires it to be a list, the method should just return the single value.